### PR TITLE
[AUD-1223] [AUD-1005] Add ChallengeRewardsDrawer and CognitoDrawer

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

~**NOTE:** This is based on #206~ Updated this PR to be based on master

AUD-1005 AUD-1223

- Adds the `ChallengeRewardsDrawer` that mimics the `ChallengeRewardsModal` on web
- Adds a new `CognitoDrawer` modal component that uses a webview (note: similar to `CognitoModal` on web, but different! see below)

Note: Follows a decent amount of changes on `audius-client` to make this smoother
- [x] **TODO:** Update client version once https://github.com/AudiusProject/audius-client/pull/794 goes in

Screenshots:
image
image

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

- This is a bit different from the existing mobile Cognito flow, which uses a separate `window.Flow` API. Instead, this one calls Identity to get an embeddable/shareable URL for use with a mobile webview. See: https://github.com/AudiusProject/audius-protocol/pull/2284
- Note that the HCaptcha does not require the challenge modal to be hidden on mobile, but I left that the same for consistency/code-reuse.
- Progress bar has _slightly_ different look, since an inset shadow isn't available in React Native. Imitated best I could.

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

- Tested on iOS simulator manually using a latest master webclient and staging backend.

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

N/A - Errors in libs for claiming rewards should get passed through to sentry

### CODE REVIEW

**Comment by rickyrombo** [comment]
Oops, this was suppose to be a draft!

File: src/components/cognito-drawer/CognitoDrawer.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ const createStyles = (themeColors: ThemeColors) =>
     }
   })
 
-const CognitoWebView = () => {
+export const CognitoDrawer = () => {
```
Comment (Kyle-Shanks) [inline src/components/cognito-drawer/CognitoDrawer.tsx:62]: named export. love it 🙏

File: src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+              </Text>
+            )}
+          </View>
+        )}
+      </View>
+    </Drawer>
+  )
+}
+
+export default ChallengeRewardsDrawer
```
Comment (Kyle-Shanks) [inline src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx:336]: should use named export here too

File: src/components/challenge-rewards-drawer/ChallengeRewardsDrawerProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+      isDisbursed={challenge.is_disbursed}
+      claimStatus={claimStatus}
+      onClaim={hasConfig ? onClaim : undefined}
+    >
+      {contents}
+    </ChallengeRewardsDrawer>
+  )
+}
+
+export default ChallengeRewardsDrawerProvider
```
Comment (Kyle-Shanks) [inline src/components/challenge-rewards-drawer/ChallengeRewardsDrawerProvider.tsx:295]: 📛 ❌ 🇵🇹

File: src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+}
+
+const createProgressBarStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    progressBarContainer: {
+      backgroundColor: themeColors.neutralLight9,
+      borderRadius: 22,
+      height: 24,
+      marginTop: 14,
+      marginBottom: 14,
```
Comment (Kyle-Shanks) [inline src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx:34]: No need to update, but I recently learned that you can use marginVertical for things like this

File: src/components/challenge-rewards-drawer/ReferralLinkCopyButton.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    },
+    copyPromptContent: {
+      padding: 16
+    },
+    copyPrompt: {
+      marginBottom: 12,
+      fontSize: 16,
+      textTransform: 'uppercase',
+      textAlign: 'center',
+      color: '#c675ff',
```
Comment (Kyle-Shanks) [inline src/components/challenge-rewards-drawer/ReferralLinkCopyButton.tsx:36]: should use a themeColor here?

**Review by Kyle-Shanks — APPROVED** [review]

File: src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx (hunk 4 above)
Comment (raymondjacobson) [reply src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx:34]: wow!

File: src/components/challenge-rewards-drawer/ProfileCompletionChecks.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+const renderArrowIcon = color => <IconArrow fill={color} />
+
+export const ProfileCompletionChecks = ({
+  isComplete,
+  onClose
+}: {
+  isComplete: boolean
+  onClose: () => void
+}) => {
+  const handle = useSelectorWeb(getUserHandle)!
```
Comment (raymondjacobson) [inline src/components/challenge-rewards-drawer/ProfileCompletionChecks.tsx:69]: can we avoid the `!`? maybe just a null/no-render check on handle?

File: src/components/challenge-rewards-drawer/ReferralLinkCopyButton.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+  const { toast } = useContext(ToastContext)
+  const onCopyClicked = useCallback(() => {
+    Clipboard.setString(referralUrl)
+    toast({ content: messages.copyNotice, type: 'info' })
+  }, [referralUrl, toast])
+
+  // Button animation
+  const scale = useRef(new Animated.Value(1)).current
+  const handlePressIn = () => {
```
Comment (raymondjacobson) [inline src/components/challenge-rewards-drawer/ReferralLinkCopyButton.tsx:75]: Maybe we should generic/hookify this one

**Review by raymondjacobson — APPROVED** [review]

File: src/components/challenge-rewards-drawer/ReferralLinkCopyButton.tsx (hunk 7 above)
Comment (rickyrombo) [reply src/components/challenge-rewards-drawer/ReferralLinkCopyButton.tsx:75]: #234 for either fast follow or to merge into this :)

File: src/components/challenge-rewards-drawer/ProfileCompletionChecks.tsx (hunk 6 above)
Comment (rickyrombo) [reply src/components/challenge-rewards-drawer/ProfileCompletionChecks.tsx]: Updated to just use `currentUser`

**Pull request merged**